nixos/doc/manual/release-notes/rl-2111.section.md

```diff
@@ -307,6 +307,23 @@ To be able to access the web UI this port needs to be opened in the firewall.
     `myhostname`, but before `dns` should use the default priority
   - NSS modules which should come after `dns` should use mkAfter.
 
+- NSS modules are now globally provided (by a `/run/nss-modules` symlink),
+  similar to how we handle OpenGL drivers.
+
+  This removes the need for nscd as a proxy for all NSS requests, and avoids
+  DNS requests leaking across network namespaces.
+
+  While doing this upgrade, existing applications need to be restarted, so
+  they know how to pick up NSS modules from `/run/nss-modules`.
+
+  If you want to defer application restart to a later time, explicitly enable
+  `nscd` via `services.nscd.enable` until the application restart.
+
+  We can mix NSS modules from any version of glibc according to
```

Comment:
What happens in the following scenario?

1. Some program P is built against glibc version A
2. System gets upgraded to glibc version B that has newer version symbols, alongside with new nss modules that now link against glibc version B
3. Program P gets started (still built against glibc A, because eg. it comes from a user profile), glibc A attempts to dlopen() nss modules built against glibc B.
4. **Guess:** dlopen() failes because nss module attempts to load glibc symbols versioned for glibc B, but the process already has glibc A in memory.

Is this a problem known with other .so's in /run, like GL?

Reply:
I _think_ the above scenario being true (including guess at 4.) is what's being implied by the following comment from the linked libc-help thread:

```
i've seen dlopen of a new libnss_dns fail because it depends on
libresolv and the libc was too old to load a newer libresolv
(causing name lookup failures on a live system after a libc update).
```

Reply:
Yes. But in that case, the NSS module will just fail, and things move on to the next NSS module in nsswitch.conf - which is probably less of a bitter pill to swallow than a somewhat flaky/sometimes caching/DNS leaking nscd.

Reply:
I agree that this shouldn't block this PR. But if it's a known issue, it should be explicitly documented in the manual.

Reply:
@q3k do you have a proposal on how to add this to the release notes?